src/hooks/useBanner.ts

```diff
@@ -0,0 +1,175 @@
+ * @param now - The reference date/time to check against (defaults to current date/time).
+ * @returns `true` if the banner is active at the given time, otherwise `false`.
+ */
+export function isBannerActive(banner: BannerConfig, now = new Date()): boolean {
+  const timeZone = banner.timeZone || 'UTC';
+  const zonedNow = convertToTimeZone(now, timeZone);
+  const nowHM = zonedNow.getHours() * 60 + zonedNow.getMinutes();
+
+  switch (inferRepeatType(banner)) {
+    case 'none': {
+      const hasDate = banner.year && banner.month && banner.dayOfMonth;
+      // If user does not sepecify the year/month/day, let us assume it is today.
+      const year = hasDate ? Number(banner.year) : zonedNow.getUTCFullYear();
+      const month = hasDate ? Number(banner.month) - 1 : zonedNow.getUTCMonth(); // 0-based
+      const day = hasDate ? Number(banner.dayOfMonth) : zonedNow.getUTCDate();
+
+      // When there is no any time specified, we assume the banner should be shown at once
+      if (!banner.startTime || !banner.endTime) return true;
+
+      const [startHour, startMinute] = banner.startTime.split(':').map(Number);
+      const [endHour, endMinute] = banner.endTime.split(':').map(Number);
+
+      const start = new Date(Date.UTC(year, month, day, startHour, startMinute));
+      const end = new Date(Date.UTC(year, month, day, endHour, endMinute));
+      const startZoned = convertToTimeZone(start, timeZone);
+      const endZoned = convertToTimeZone(end, timeZone);
+
+      return zonedNow >= startZoned && zonedNow <= endZoned;
+    }
+
+    case 'weekly': {
+      if (zonedNow.getDay() !== banner.dayOfWeek) return false;
+      const startHM = parseHM(banner.startTime);
+      const endHM = parseHM(banner.endTime);
+      return nowHM >= startHM && nowHM <= endHM;
+    }
+
+    case 'monthly': {
+      if (zonedNow.getDate() !== banner.dayOfMonth) return false;
+      const startHM = parseHM(banner.startTime);
+      const endHM = parseHM(banner.endTime);
+      return nowHM >= startHM && nowHM <= endHM;
+    }
+
+    default:
+      return false;
+  }
+}
+
+export const useBanner = () => {
+  const {
+    data: bannerYamlData,
+    isLoading,
+    error,
+  } = useK8sWatchResource<ConfigMap>(
+    {
+      groupVersionKind: ConfigMapGroupVersionKind,
+      namespace: 'konflux-info',
+      name: 'konflux-banner-configmap',
+      watch: true,
+    },
+    ConfigMapModel,
+  );
+
+  return useMemo(() => {
+    if (isLoading || error) return null;
+
+    const yamlContent = bannerYamlData?.data?.['banner-content.yaml'];
+    if (typeof yamlContent !== 'string') return null;
+
+    const bannerList = parseBannerList(yamlContent);
+
+    // Check from last to first and return the latest active one
+    for (let i = bannerList.length - 1; i >= 0; i--) {
+      const banner = bannerList[i];
+      if (isBannerActive(banner)) {
+        return { type: banner.type, summary: banner.summary };
+      }
+    }
```

Comment:
why do you want to check in reverse order, also use array.find to get the active banner

Reply:
For the reverse:
 I am considering the following scenario:
We have one weekly banner for regular maintenance. it would be shown between 10:00AM - 14:00AM today.
Meanwhile, another system issue A happens during the two hours and users need to add one  alert.

The user just need to append one banner to the list and do not need to check about others. 
UI consider the latest active banner A is the one we need to show up and show it with the its specified time.
When A is obsoleted, the existing regular maintenance banner would be shown again automatically.

If users insert new banners as the first one in the banner list, then i do not need to reverse it.
So in short, which way we prefer users to add more content? insert to top or append to end?


For the find, would update.

Reply:
updated